pkg/apis/pipeline/v1alpha1/verificationpolicy_validation.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	"github.com/tektoncd/pipeline/pkg/apis/validate"
+	"knative.dev/pkg/apis"
+)
+
+var _ apis.Validatable = (*VerificationPolicy)(nil)
+
+// Validate VerificationPolicy
+func (v *VerificationPolicy) Validate(ctx context.Context) (errs *apis.FieldError) {
+	errs = errs.Also(validate.ObjectMetadata(v.GetObjectMeta()).ViaField("metadata"))
+	errs = errs.Also(v.Spec.Validate(ctx))
+	return errs
+}
+
+// Validate VerificationPolicySpec
+func (vs *VerificationPolicySpec) Validate(ctx context.Context) (errs *apis.FieldError) {
+	if len(vs.Resources) == 0 {
+		errs = errs.Also(apis.ErrMissingField("resources"))
```

Comment:
does it make sense to validate the Resource field as well? e.g. what if the patterns are blank or invalid?

Reply:
yes that makes sense, we can validate it by compiling the pattern

Reply:
That's a good point! Thanks for the suggestion! I think we can call regexp.Compile to validate it

Reply:
fixed latest commit